Documentation for mocking module

elihart opened the pull request:
This fleshes out the mocking documentation as well as upstreams some improvements to the mocking infra we have made internally the last few months.

gpeal reviewed (approved):
Thanks for doing this! What is the plan for Airbnb to move to 2.0?

elihart commented on mock_generation/package:
```diff
@@ -0,0 +1,5 @@
+# Run this script any time there are changes to the mock printer script.
+# It will repackage the binary and make sure it is given executable permissions.
+kscript --package MvrxMockPrinter.kts
+chmod +x MvrxMockPrinter
```
yep, will do

elihart commented:
@gpeal I have bumped the kotlin version and some other dependencies, and removed jetifier (which should speed up builds)

I also did all of the mvrx -> mavericks renames for the mocking and launcher modules (besides the package names), and renamed install to initialize for MockableMavericks
